backends-clickhouse/src/main/java/org/apache/gluten/vectorized/CHNativeBlock.java

```diff
@@ -111,4 +117,14 @@ public ColumnarBatch toColumnarBatch() {
     }
     return new ColumnarBatch(vectors, numRows);
   }
+
+  public static ColumnarBatch slice(ColumnarBatch batch, int offset, int limit) {
+    if (limit >= batch.numRows()) {
```

Comment:
what if both limit and offset exceed batch.numRows() ?

Reply:
if (offset >=   batch.numRows())  throw exception; 
limit = min(limit, batch.numRow() - offset);